icepyx/core/granules.py

```diff
@@ -462,25 +490,34 @@ def download(self, verbose, path, session=None, restart=False):
             if verbose is True:
                 print("Zip download URL: ", downloadURL)
             print("Beginning download of zipped output...")
-            zip_response = session.get(downloadURL)
-            # Raise bad request: Loop will stop for bad response code.
-            zip_response.raise_for_status()
-            print(
-                "Data request",
-                order,
-                "of ",
-                len(self.orderIDs[i_order:]),
-                " order(s) is downloaded.",
-            )
+            
+            try:
+                zip_response = session.get(downloadURL)
+                # Raise bad request: Loop will stop for bad response code.
+                zip_response.raise_for_status()
+                print(
+                    "Data request",
+                    order,
+                    "of ",
+                    len(self.orderIDs[i_order:]),
+                    " order(s) is downloaded.",
+                )
+                success = True
+            except requests.HTTPError:
+                print(
+                    "Unable to download ",
+                    order,
+                    ". Check granule order for messages.")
+                success = False
 
             # DevGoal: move this option back out to the is2class level and implement it in an alternate way?
             #         #Note: extract the dataset to save it locally
-            # if extract is True:
-            with zipfile.ZipFile(io.BytesIO(zip_response.content)) as z:
-                for zfile in z.filelist:
-                    # Remove the subfolder name from the filepath
-                    zfile.filename = os.path.basename(zfile.filename)
-                    z.extract(member=zfile, path=path)
+            if success is True:
```

Comment:
Instead of the `success` variable, you could add an `else` block in the try-catch:

```
try:
...
except:
...
else:
    with zipfile.ZipFile(io.BytesIO(zip_response.content)) as z:
        for zfile in z.filelist:
            # Remove the subfolder name from the filepath
            zfile.filename = os.path.basename(zfile.filename)
            z.extract(member=zfile, path=path)
```

Reply:
Change made - thanks for the suggestion!